sklearn/cluster/_hdbscan/_linkage.pyx

```diff
@@ -194,25 +196,28 @@ cpdef cnp.ndarray[MST_edge_t, ndim=1, mode='c'] mst_from_data_matrix(
                 next_node_core_dist,
                 pair_distance
             )
-            if mutual_reachability_distance > next_node_min_reach:
-                if next_node_min_reach < new_reachability:
-                    new_reachability = next_node_min_reach
-                    source_node = next_node_source
-                    new_node = j
-                continue
 
+            # If MRD(i, j) is smaller than node j's min_reachability, we update
+            # node j's min_reachability for future reference.
             if mutual_reachability_distance < next_node_min_reach:
                 min_reachability[j] = mutual_reachability_distance
                 current_sources[j] = current_node
+
+                # If MRD(i, j) is also smaller than node i's current
+                # min_reachability, we update and set their edge as the current
+                # MST edge candidate.
                 if mutual_reachability_distance < new_reachability:
                     new_reachability = mutual_reachability_distance
                     source_node = current_node
                     new_node = j
-            else:
-                if next_node_min_reach < new_reachability:
-                    new_reachability = next_node_min_reach
-                    source_node = next_node_source
-                    new_node = j
+                continue
+
+            # If the node j is closer to another node already in the tree, we
+            # make their edge the current MST candidate edge.
+            if next_node_min_reach < new_reachability:
```

Comment:
I have a very small +0.05 preference with using `else if` here:

```suggestion
            # If the node j is closer to another node already in the tree, we
            # make their edge the current MST candidate edge.
            else if next_node_min_reach < new_reachability:
```

(I think of `continue` as an early exit so the code inside should be "simple".)

I am okay either way.

Reply:
Yeah actually I agree with you here, it is a bit nicer. Done.